### ISSUE TITLE
Possible string escape bug when including CLI11 via FetchContent

### ISSUE THREAD

**Issue by Dr-ZeeD** [CLIUtils/CLI11#457]
When trying to include CLI11 via FetchContent, a dependent option is used in CMake to determine whether the documentation should be built.
Part of this is [checking whether the `docs` directory exists.](https://github.com/CLIUtils/CLI11/blob/master/CMakeLists.txt#L49).
At the [evaluation of all conditions](https://github.com/CLIUtils/CLI11/blob/master/CMakeLists.txt#L59), escaping of spaces might be off.

Say my project is located at `/some/directory with spaces`. When including CLI11 via
```
FetchContent_Declare(
    clilib
    GIT_REPOSITORY https://github.com/CLIUtils/CLI11.git
    GIT_TAG v1.9.0
)
FetchContent_MakeAvailable(clilib)
```
CMake gives me the following error during `cmake ..`
```
CMake Error at /path_to_cmake/3.17.0_1/share/cmake/Modules/CMakeDependentOption.cmake:33 (if):
  if given arguments:

    "EXISTS" "/some/directory" "with" "spaces/build/_deps/clilib-src/docs"

  Unknown arguments specified
Call Stack (most recent call first):
  build/_deps/clilib-src/CMakeLists.txt:57 (cmake_dependent_option)
```

From the error messages I reckon there is an error with escaping the spaces.

**Comment by henryiii** [comment]
I think we can add quotes `\"` around the checks in the strings.

**Comment by henryiii** [comment]
Can you test #471?

**Comment by Dr-ZeeD** [comment]
I did, unfortunately it still does not work.
CMake seems to be rather particular when it comes to string escaping in commands.

The workaround I have in place at the moment is rather crude:

```
if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/docs")
    set(docs_EXIST TRUE)
else()
    set(docs_EXIST FALSE)
endif()
list(APPEND build-docs "docs_EXIST")
```

**Comment by henryiii** [comment]
I've tried something more like that which is in examples. Does it work now?

**Comment by Dr-ZeeD** [comment]
It still does not work. CMake seems to do rather unexpected things when trying to lazily evaluate this by constructing a list via strings... 🤔

**Comment by henryiii** [comment]
Okay, will try again with something that looks exactly like your example in a day or two.

**Comment by henryiii** [comment]
One more time, can you test? This should be pretty much exactly your suggestion, so it should work.

**Comment by Dr-ZeeD** [comment]
Works like a charm! Thank you very much and please excuse my delayed answer – I have not found time until now.

**Comment by henryiii** [comment]
Merged, tagged, and released v1.9.1!
